dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/IntegratorVariableLimits.mo

```diff
@@ -0,0 +1,96 @@
+within Dynawo.NonElectrical.Blocks.Continuous;
+
+/*
+* Copyright (c) 2022, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite
+* of simulation tools for power systems.
+*/
+
+model IntegratorVariableLimits "Integrator with limited value of output (variable limits) and freeze"
+  import Modelica;
+  import Dynawo.Types;
+
+  extends Modelica.Blocks.Interfaces.SISO;
+
+  parameter Boolean DefaultLimitMax = true "If limitMin > limitMax : if true, y = limitMax, if false, y = limitMin";
+  parameter Types.PerUnit K = 1 "Integrator gain";
+  parameter Types.PerUnit Tol = 0.01 "Tolerance on limit crossing";
+
+  Modelica.Blocks.Interfaces.RealInput limitMax "Connector of Real input signal used as maximum of input u" annotation(
+    Placement(transformation(extent={{-140,60},{-100,100}})));
+  Modelica.Blocks.Interfaces.RealInput limitMin "Connector of Real input signal used as minimum of input u" annotation(
+    Placement(transformation(extent={{-140,-100},{-100,-60}})));
+
+  Types.PerUnit v "Integrator input not accounting for the freeze";
+  Types.PerUnit w(start = Y0) "Non-limited integrator output";
+
+  parameter Types.PerUnit Y0 = 0 "Initial or guess value of output (must be in the limits limitMin .. limitMax)" annotation(
+    Dialog(group="Initialization"));
+
+equation
+  v = K * u;
+
+  der(w) = v * (1 - tanh((w - limitMax + Tol) / Tol)) * (1 - tanh((limitMin + Tol - w) / Tol)) / 4;
```

Comment:
I am not a big fan of the use of this tanh. Could you try to go back to a version where you implement the same block as in Eurostag ? We will then try to figure out what's not working in the test case. Thanks